packages/fast-components-styles-msft/src/button/index.ts

```diff
@@ -43,33 +70,111 @@ function applyTransaprentBackground(): ICSSRules<IDesignSystem> {
+    const focusBoxShadowDefaults: string = "inset 0 0 0 2px";
+
+    // Define secondary button colors
+    const color: string = "white";
+    const secondaryBackgroundColor: string = scaledEnsureNormalContrast(
+        scaledNormalContrast(backgroundColor, foregroundColor),
+        color
+    );
+    const secondaryHoverBackgroundColor: string = adjustContrast(
+        ContrastModifiers.hover,
+        secondaryBackgroundColor,
+        color
+    );
+    const secondaryFocusBorderColor: string = scaledEnsureNormalContrast(
+        scaledEnsureNormalContrast(foregroundColor, backgroundColor),
+        secondaryBackgroundColor
+    );
+
+    const secondaryFocusBoxShadow: string = Chroma.contrast(
+        secondaryBackgroundColor,
+        secondaryFocusBorderColor
+    ) < scaleContrastNormal(contrastScale)
+        ? `${focusBoxShadowDefaults} ${ensureNormalContrast(contrastScale, secondaryBackgroundColor, secondaryFocusBorderColor)}`
+        : "none";
+    const secondaryDisabledBackgroundColor: string = adjustContrast(
+        ContrastModifiers.disabled,
+        secondaryBackgroundColor,
+        color
+    );
+
+    // Define primary button colors
+    const primaryRestBackgroundColor: string = scaledEnsureNormalContrast(
+        scaledEnsureNormalContrast(brandColor, backgroundColor),
+        color
+    );
+    const primaryHoverBackground: string = adjustContrast(
+        ContrastModifiers.hover,
+        primaryRestBackgroundColor,
+        color
+    );
+    const primaryFocusBorderColor: string = scaledEnsureNormalContrast(
+        scaledEnsureNormalContrast(foregroundColor, backgroundColor),
+        primaryRestBackgroundColor
+    );
+    const primaryFocusBoxShadow: string = Chroma.contrast(
+        primaryRestBackgroundColor,
+        primaryFocusBorderColor
+    ) < scaleContrastNormal(contrastScale)
+        ? `${focusBoxShadowDefaults} ${ensureNormalContrast(contrastScale, primaryRestBackgroundColor, primaryFocusBorderColor)}`
+        : "none";
+    const primaryDisabledBackground: string = adjustContrast(
+        ContrastModifiers.disabled,
+        primaryRestBackgroundColor,
+        backgroundColor
+    );
+    const primaryDisabledColor: string = contrast(
+        ContrastModifiers.disabled * -1,
+        color,
+        primaryDisabledBackground
+    );
+
+    const outlineColor: string = scaledEnsureNormalContrast(foregroundColor, backgroundColor);
+    const outlineBorderColor: string = scaledNormalContrast(foregroundColor, backgroundColor);
+    const outlineDisabledColor: string = contrast(
+        ContrastModifiers.disabled * -1,
+        outlineColor,
+        backgroundColor
+    );
+    const outlineDisabledBorderColor: string = outlineDisabledColor;
 
     return {
         button: {
             ...applyType("t7", "vp1"),
             boxSizing: "border-box",
-            maxWidth: toPx(374),
-            minWidth: toPx(120),
+            maxWidth: "374px",

```

Comment:
Why is did you remove toPx?  Just curiousl

Reply:
Its an unnecessary operation that needs to happen at runtime. `toPx` is useful when number manipulation is needed, but if we know the value already there is no reason I know of to take the runtime hit - just define it as a string.